CheckboxItem/Checkbox.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Agate styled checkbox component.
+ *
+ * @example
+ * <Checkbox onToggle={console.log} />
+ *
+ * @module agate/Checkbox
+ * @exports Checkbox
+ * @exports CheckboxBase
+ */
+
+import kind from '@enact/core/kind';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import Icon from '../Icon';
+
+import ToggleIcon from './ToggleIcon';
+
+import css from './Checkbox.module.less';
+
+/**
+ * A checkbox component, ready to use in Agate applications.
+ *
+ * `Checkbox` may be used independently to represent a toggleable state but is more commonly used as
+ * part of [CheckboxItem]{@link agate/CheckboxItem}.
+ *
+ * Usage:
+ * ```
+ * <Checkbox selected />
+ * ```
+ *
+ * @class Checkbox
+ * @memberof agate/Checkbox
+ * @extends agate/ToggleIcon.ToggleIcon
+ * @ui
+ * @private
+ */
+const CheckboxBase = kind({
```

Comment:
Let's move this to a top-level component like `Switch`. `ToggleIcon` might be a useful private top-level or `internal` component to share between `Switch` and `Checkbox` too. Please check that out and decide if it's useful or not.